### PR TITLE
Child Mode

### PR DESCRIPTION
**Pull request by jvc56**

### CODE REVIEW

File: pkg/stores/user/db.go [hunk 1]
```diff
@@ -467,8 +470,9 @@ func (s *DBStore) SetPersonalInfo(ctx context.Context, uuid string, email string
 			return result.Error
 		}
 
-		return tx.Model(p).Update(map[string]interface{}{"first_name": firstName,
+		return s.db.Model(p).Update(map[string]interface{}{"first_name": firstName,
```
Comment (domino14) [inline pkg/stores/user/db.go:473]: how come not tx?

Comment (jvc56) [reply pkg/stores/user/db.go:473]: Oops, that was an oversight.

**Comment by atomiccupcake** [comment]
Front end will warn users once a day if they haven't submitted a birthdate. Will hide profile info and full names for children and users of unknown age on the profile. Games will no longer show full names for anyone. (Making a new issue to correct.)

File: liwords-ui/src/socket/socket.ts [hunk 2]
```diff
@@ -107,10 +109,26 @@ export const LiwordsSocket = (props: {
...
+          isChild: decoded.cs,
           path: pathname,
           perms: decoded.perms?.split(','),
         },
       });
+      if (
+        parseInt(decoded.cs) === 2 &&
+        (!localStorage?.getItem('birthdateWarning') ||
+          Date.now() - parseInt(localStorage?.getItem('birthdateWarning')!) >
+            24 * 3600)
```
Comment (andy-k) [inline liwords-ui/src/socket/socket.ts:121]: isn't `Date.now()` in milliseconds?
(also it'd be preferable not to `localStorage.getItem` twice)

File: liwords-ui/src/socket/socket.ts [hunk 3]
```diff
@@ -114,11 +114,12 @@ export const LiwordsSocket = (props: {
...
       });
+      const bdateWarning = localStorage?.getItem('birthdateWarning');
       if (
         parseInt(decoded.cs) === 2 &&
-        (!localStorage?.getItem('birthdateWarning') ||
-          Date.now() - parseInt(localStorage?.getItem('birthdateWarning')!) >
-            24 * 3600)
+        decoded.a &&
+        (!bdateWarning ||
+          Date.now() - parseInt(bdateWarning) > 24 * 3600 * 1000)
```
Comment (atomiccupcake) [inline liwords-ui/src/socket/socket.ts:122]: Nice catch

File: pkg/profile/profile.go [hunk 4]
```diff
@@ -78,7 +78,7 @@ func (ps *ProfileService) GetProfile(ctx context.Context, r *pb.ProfileRequest)
 		FirstName:       user.Profile.FirstName,
```
Comment (atomiccupcake) [inline pkg/profile/profile.go:78]: First and last name should also be suppressed if youth

**Review by atomiccupcake — APPROVED** [review]

File: pkg/profile/profile.go (hunk 4 above)
Comment (atomiccupcake) [reply pkg/profile/profile.go:78]: And the "About"

File: liwords-ui/src/settings/personal_info.tsx [hunk 5]
```diff
@@ -298,6 +309,34 @@ export const PersonalInfo = React.memo((props: Props) => {
...
+              <>
+                Date of birth{' '}
+                <span className="notice">(This will not be displayed.)</span>
+              </>
+            }
+            rules={[
+              {
+                required: true,
+                message:
+                  'Your profile information will be private unless you provide a birthdate.',
```
Comment (andy-k) [inline liwords-ui/src/settings/personal_info.tsx:326]: if DOB is mandatory, this message should be reworded - this message seems to say it's fine to have a private bio

**Pull request merged**